src/main/java/com/bp3x/raidbot/commands/lfg/DeleteCommand.java

```diff
@@ -1,62 +1,59 @@
-            "`!delete 5241`";
-
     public DeleteCommand() {
         this.name = "delete";
-        this.help = DELETE_HELP_EXAMPLE;
-        this.arguments = "<eventID>>";
-        this.guildOnly = false;
+        this.help = "\nUse to delete an event by providing the event ID.\n\nExample:\n`/delete 5241`";
+        this.guildOnly = true;
         this.ownerCommand = false;
+        
+        this.options = Collections.singletonList(new OptionData(OptionType.INTEGER, "event-id", "The ID # of the event you want to delete.", true));
     }
 
     @Override
-    protected void execute(CommandEvent commandEvent) {
-
-        String[] args = commandEvent.getArgs().split("\\s+");
-
-        if (args.length == 1) {
-            final Event eventToDelete = Event.getEventById(args[0]);
+    protected void execute(SlashCommandEvent commandEvent) {
+        commandEvent.deferReply(true).queue(hook -> {
+            var eventId = commandEvent.optString("event-id");
+            
+            final Event eventToDelete = Event.getEventById(eventId);
             if (eventToDelete != null) {
                 final Message messageToDelete = Event.findMessageFromEvent(eventToDelete);
-                commandEvent.getChannel().retrieveMessageById(messageToDelete.getId()).queue(latestMessageInstance ->
-                {
-                    if (latestMessageInstance != null) {
-                        ThreadChannel thread = latestMessageInstance.getStartedThread();
-                        if (thread != null) {
-                            thread.delete().queue();
-                            messageToDelete.delete().queue();
-                            MessageUtils.sendAutoDeletedMessage("Deleted event with ID: " + eventToDelete.getEventId() + " and its corresponding thread.", 300, commandEvent);
-                        } else {
-                            messageToDelete.delete().queue();
-                            MessageUtils.sendAutoDeletedMessage("Deleted event with ID: " + eventToDelete.getEventId() + ". Please delete the thread manually, there was a problem deleting it.", 300, commandEvent);
-                        }
-                    } else {
-                        MessageUtils.sendAutoDeletedMessage("There was a problem locating the message to delete the thread. Contact a mod.", 300, commandEvent);
-                    }
-                });
+                commandEvent.getChannel().retrieveMessageById(messageToDelete.getId()).queue(latestMessageInstance -> {
+                     if (latestMessageInstance != null) {
+                         ThreadChannel thread = latestMessageInstance.getStartedThread();
+                         if (thread != null) {
+                             thread.delete().queue();
+                             messageToDelete.delete().queue();
+                             hook.editOriginal("Deleted event with ID: " + eventToDelete.getEventId() + " and its corresponding thread.").queue();
```

Comment:
Do these hooks throughout this command auto delete like previous messages to help reduce clutter in the channel where you interact with the bot?

Reply:
No, because we're using `commandEvent.deferReply(true)`. We have to use `deferReply()` because you only have 3 seconds to respond to the command. `deferReply()` makes it look like the bot is "typing" and lets us respond in a message later (with `hook.editOriginal()`). 

When we pass `true` as a parameter for `deferReply()`, it determines whether the bot's response is ephemeral or not. The bot's response is not visible to anyone except for the person who executed the command, and that user can manually dismiss the message once they have read it. 
<img width="434" alt="image" src="https://user-images.githubusercontent.com/9632473/227738574-f81883c8-4a9d-4317-89f8-5fec2226713d.png">